homeassistant/components/lifx/light.py

```diff
@@ -424,6 +433,26 @@ def unregister(self, bulb):
             entity.registered = False
             entity.async_write_ha_state()
 
+    async def entity_registry_updated(self, event):
+        """Handle entity registry updated."""
+        if event.data["action"] == "remove":
+            await self.remove_empty_devices()
+
+    async def remove_empty_devices(self):
+        """Remove devices with no entities."""
+        entity_reg = await er.async_get_registry(self.hass)
+        device_reg = dr.async_get(self.hass)
+        device_list = dr.async_entries_for_config_entry(
+            device_reg, self.config_entry.entry_id
+        )
+        for device_entry in device_list:
+            if not er.async_entries_for_device(
+                entity_reg,
+                device_entry.id,
+                include_disabled_entities=True,
+            ):
+                device_reg.async_remove_device(device_entry.id)
```

Comment:
Maybe only remove the config entry id from this device? If it's the only config entry, the device will be removed.

Reply:
I am not sure how to do that. Anyway, for LIFX I think it does not matter since there is 1 config entry, N devices and 1 entity per device?

Reply:
It could matter if another integration has the device mapped via the mac address, but that probably isn't a problem here since LiFX is the main integration of these devices.

Reply:
I see. Can you give me a hint on how to implement your proposal?

Reply:
Pass the config entry_id that you want to remove from the device entry to `async_update_device`.

https://github.com/home-assistant/core/blob/c93d389544e075fbb567acfe26067a96730963e2/homeassistant/helpers/device_registry.py#L414